src/libcmd/installables.cc

```diff
@@ -514,6 +514,61 @@ ref<Installable> SourceExprCommand::parseInstallable(
     return installables.front();
 }
 
+static StorePath getBuiltPathFromRealisation(
+    Store & store,
+    const std::map<std::string, Hash> & outputHashes,
+    const DerivationOutputsAndOptPaths & drvOutputs,
+    const StorePath & drvPath,
+    const std::string & output)
+{
+    if (!outputHashes.count(output))
+        throw Error(
+            "the derivation '%s' doesn't have an output named '%s'",
```

Comment:
Would this be more accurate, or is the original message better, e.g. in case the output is also missing from `drvOutputs`?

```suggestion
            "the realisation for derivation '%s' doesn't have an output named '%s'",
```